### PR TITLE
[components] Break dagster-components into separate package

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Breaks components code into a new `dagster-components` package that will be published independently of the main Dagster release cycle. For now, `dagster-components` also includes the `dg` CLI tool.

Because this is placed in `examples/experimental` rather than a directory containing by-default published packages, no modifications are required to the release machinery to prevent its publication.

## How I Tested These Changes

Existing test suite after moving everything over.

### CODE REVIEW

**Comment by smackesey** [comment]
* **#26237** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26237?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26237?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: examples/experimental/dagster-components/setup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    name="dagster-components",
+    version=get_version(),
+    author="Dagster Labs",
+    author_email="hello@dagsterlabs.com",
+    license="Apache-2.0",
+    description="",  # TODO - fill out description
+    url=(
+        "https://github.com/dagster-io/dagster/tree/master/examples/experimental/dagster-components"
+        "dagster-components"
+    ),
```
[the remark this answers is not in the document]
Comment (schrockn) [reply examples/experimental/dagster-components/setup.py:28]: this looks real

**Comment by schrockn** [comment]
Got it. So for now our bootstraping procedure will be

`uv tool install dagster-component`
`dg generate deployment my_deployment`

**Review by schrockn — APPROVED** [review]
Ok.

Next we will need scripts for pushing this up to `pypi` independently.

@dpeng817 or @benpankow should be able to help as they have set this up. Ideal would be a playbook plus some common code so we don't have to do this from scratch every time.

**Pull request merged**